### DISCUSSION TITLE
distrobox keeps my terminal host name the same when entering a container.

### DISCUSSION THREAD

**Discussion by bitzorbites** [89luca89/distrobox#1537] [Q&A]
in some places I'm reading this seems like the correct behavior, however, every video I've watch shows the hostname part of the shell prompt change to the name of the container.  

expected behavior: entering a distrobox causes the hostname portion of my shell prompt to change to containers host name.
example:  user@originalhostname becomes user@ubuntucontainer1 when entering the container named ubuntucontainer1

actual behavior: my user deck@steamdeck remains deck@steamdeck even while inside a new ubuntu container. using the command apt proves my container does work, because my host is an arch based distro.

screenshot of terminal creating and entering an ubuntu-20-04 container:

![hostname](https://github.com/user-attachments/assets/94bb1982-c8be-4f54-ae4f-e55587717df7)

screenshot from a popular youtube channel using distrobox:
![youtubedistrobox](https://github.com/user-attachments/assets/7f21470a-c0fb-4d32-8a11-bc959626b987)

This is very confusing behavior, and its quite hard to tell when you are actually inside of a container. for almost a day, i thought distrobox just wasnt working at all until I tried a simple apt command and realized I was actually in a container even though the prompt wasn't changed.

i am using a steamdeck with manual installation of latest distrobox and podman-launcher in /home/deck/.local/bin and added to the $PATH variable.    on a side note, as a second issue, i had to remove the "xhost +si:localuser:$USER" from .bashrc because everytime I created a container it would say "xhost: command not found" So I have been unable to get any gui applications to work.

**Comment by ivenomweed** [comment]
Hey I was having the same issue, 

I know this was 2 months a go, But if someone else is facing the same issue please try with the `--hostname` something like `distrobox create --name arch3 --hostname arch3 --image quay.io/toolbx/arch-toolbox:latest` IDK why but it's not set by default.

**Comment by tgf9** [comment]
I was having the same issue. I realized that the default behavior is to use the host's hostname.

Ran this on my laptop:
```
$ distrobox create --help
--hostname:		hostname for the distrobox      default: $MY_HOSTS_HOSTNAME
```

Kinda weird because the [online help](https://distrobox.it/usage/distrobox-create/) says this:
```
--hostname:		hostname for the distrobox      default: <container-name>.$(uname -n)
```

Anyway, yeah, you need to manually specify `create --hostname XXX` to set the hostname in the container.

distrobox: 1.8.1.2
